### ISSUE TITLE
Wondering if AlDente is indeed decreasing charging cycles on my machine

### ISSUE THREAD

**Issue by GJRobert** [davidwernhart/AlDente#418]
Model: MacBook Air 2020 (M1)
OS: 11.6

Have installed and used AlDente for 1 and half month, but charging cycles count seems to keep increasing in a speed as before, not slowing down my charging cycles. But AlDente is indeed limiting my charging at 80%. I'm just wondering that, since I have keep my battery at 80% with power cable connected, why is charging cycles still increasing as before (approximately 1 cycle per day still)?

**Comment by GJRobert** [comment]
@PearApple151 I should have clarify: yes I understand and agree with what you mean. I forgot to mention that: the reason I was wondering, was because I have plugged in the power cable to keep my machine at 80% as much as possible since using AlDente. I thought, my power cycles figure can increase much slower than before, thus increase the battery life span, since AlDente has frozen my battery at 80% with external power used, so the battery charge cycle *should* (as I thought) stop increase since the laptop has stopped "charging" in this sense, and indeed the system displays that it's "not charging" all the time.

But the my charge cycle number does not increase slower as I thought! Any insights?

**Comment by xawill** [comment]
Also interested to learn more about this. My goal (and @GJRobert as well, if I understood correctly) is to keep the power adapter connected all the time and when the battery is at say 80%, power the computer directly through the power adapter, bypassing the battery. The battery is then very slowly discharging, of course, and AlDente would recharge it some time to time to keep the battery at 80%, but the large majority of the power needs of the computer would be provided directly from the power adapter.

Is it how AlDente currently works ? If yes, why are we still seeing cycles increase at the same speed as before using AlDente? If no, is it possible to have this behavior ?

**Comment by GJRobert** [comment]
Very clearly described, thank you @xawill

**Comment by PearApple151** [comment]
> Also interested to learn more about this. My goal (and @GJRobert as well, if I understood correctly) is to keep the power adapter connected all the time and when the battery is at say 80%, power the computer directly through the power adapter, bypassing the battery. The battery is then very slowly discharging, of course, and AlDente would recharge it some time to time to keep the battery at 80%, but the large majority of the power needs of the computer would be provided directly from the power adapter.
> 
> Is it how AlDente currently works ? If yes, why are we still seeing cycles increase at the same speed as before using AlDente? If no, is it possible to have this behavior ?

@GJRobert 

ok, so let me try again. you've got your computer plugged in all the time, but the cycles are still going up at the same rate as before.

My question is, has your behavior changed since you installed Al Dente?

Here's my response based on what might be a misunderstanding of your habits. When you plug the computer in, and it charges to 100%, it should bypass the battery and the computer will run on the wall power. this is with and without Al Dente on a modern machine. 

If you read about the Sailing Mode feature, and it's not the only place I've read about this, your battery will still lose charge to 2 main reasons:
1. Batteries lose charge on their own over time. maybe slowly, but still, it happens.
2. If your computer is doing something high performance and requires more power than the wall can provide, then it will draw that extra power from the battery. with a modern machine, this ought not be too much of a problem. the provided adapter should provide plenty enough juice. but if you're using an underpowered adapter, you might still notice this. 

Not an expert, this is just according my own researches. 

My point is that, if your behavior with Al Dente is the same as it was without, you ought to be using the same number of cycles. if you are keeping your computer plugged in all the time and the charge cycle was going up without Al Dente, and you're using your computer the same way with with Al Dente, then you're ought to be draining the battery just as much as you were before. the difference is that constantly charging 79-80% 100x (or 70-80 10x if you use Sailing mode) is better for the battery than 99-100 100x. 

If your computer was always plugged in before Al Dente and the cycles were going up, then it's because your computer was having to dip into the battery (or something is wrong). if your behavior hasn't changed, then your computer still needs to dip into the battery at 80%, it's just healthier for the battery at 80%.

**Comment by GJRobert** [comment]
Hi @PearApple151 thanks a lot for discussing.

Allow me explain:

- **My behavior before installing+using AlDente**: I **did not plug the cable into my MacBook all the time**, only plugged in when battery remains few (ideally plug in upon 20%, but sometimes upon < 5% if I was lazy to do it). And usually I unplug the cable once the battery reaches 100%. All these behaviors are because I have read that it's better not keep the battery at 100% all the time, but without some controlling like AlDente does, it's very hard to prevent battery at 100% while plugging in the cable very often. So I had to choose the other way, to begin to charge at least at 20% or so, and to discharge immediately upon 100% (both what I could only do). And, under this behavior, my battery charging cycles increase fast (almost 1 per day, because I use 8~10 hrs per day).
- **My behavior since using AlDente**: since AlDente can keep battery at 80% when plugged in but without charging up, so I began to **keep MacBook plugged as often as possible, keeping the laptop at 80%** (drastic change isn't it), because I thought this way the charging cycles would not increase as fast as before, thus to slow down the exhaustion of battery max capacity and battery health (hopefully).

But as I have mentioned in the first post, I was a bit stunned when I see after ~50 days of using AlDente, the increase speed of charging cycles does seem to slow down. After calculation, it still increase at roughly a pace of 1 or only fewer cycle per day. So this was why I am wondering: I thought AlDente can keep my charging cycle stop when it prevent the battery from going up beyond 80%, while using AC power as much as possible without touching the battery!

**Comment by GJRobert** [comment]
@davidwernhart Do you have an insight on this issue?
My charging cycle number is increasing day by day, even when I plug in the cable most of the time with the help of AlDente to keep battery always at 80%.

Or, battery health can indeed still be greatly maintained even with large charging cycle number?

**Comment by PearApple151** [comment]
> Hi @PearApple151 thanks a lot for discussing.
> 
> Allow me explain:
> 
> * **My behavior before installing+using AlDente**: I **did not plug the cable into my MacBook all the time**, only plugged in when battery remains few (ideally plug in upon 20%, but sometimes upon < 5% if I was lazy to do it). And usually I unplug the cable once the battery reaches 100%. All these behaviors are because I have read that it's better not keep the battery at 100% all the time, but without some controlling like AlDente does, it's very hard to prevent battery at 100% while plugging in the cable very often. So I had to choose the other way, to begin to charge at least at 20% or so, and to discharge immediately upon 100% (both what I could only do). And, under this behavior, my battery charging cycles increase fast (almost 1 per day, because I use 8~10 hrs per day).
> * **My behavior since using AlDente**: since AlDente can keep battery at 80% when plugged in but without charging up, so I began to **keep MacBook plugged as often as possible, keeping the laptop at 80%** (drastic change isn't it), because I thought this way the charging cycles would not increase as fast as before, thus to slow down the exhaustion of battery max capacity and battery health (hopefully).
> 
> But as I have mentioned in the first post, I was a bit stunned when I see after ~50 days of using AlDente, the increase speed of charging cycles does seem to slow down. After calculation, it still increase at roughly a pace of 1 or only fewer cycle per day. So this was why I am wondering: I thought AlDente can keep my charging cycle stop when it prevent the battery from going up beyond 80%, while using AC power as much as possible without touching the battery!

Sorry I was away for so long. I don't frequently visit GitHub, you took a bit of time to respond to me. haha. But I'm here again.

I suppose once upon a time it was not advisable to leave your laptop plugged in all the time. and the reason for that is what "sailing mode" exists for in the Al Dente app. You can read more about it in the help guides, but essentially now your charger should be able to handle any power that your laptop needs, so your laptop should rarely ever have to dip into the battery. So you're laptop plugged in all the time shouldn't be constantly cycling 99-100 which is the fear of leaving it plugged in. Just so you know for your other devices, just leave them plugged in. Whatever damage you might do to the battery by keeping it plugged in, is not as bad as charging it 20-100% everyday. 
These days you really only need to worry about making sure the battery gets a bit of exercise each month, and that's not hard at all. 

Whew, so your pre-al dente usage was to plug it in at 20% and charge to 100%. unplug and repeat. and you were using about a cycle a day. I can see how that caused you to use many cycles.

Now since AlDente, you leave it plugged in as much as possible, and you're still using the same cycles (ish) per day. that just doesn't make sense. I can tell you my cycles have slowed down, and my battery degradation. Something is wrong. either your math, or the app.

How are you measuring the number of cycles?

If I were you I would get this app: Better Battery 2 (or one like it). it shows when you're charging, when you're on hold, etc. I find it to be more accurate than the AlDente icon. Sometimes the AlDente icon will say I'm not charging, and the Better Battery 2 icon will say, yes you are (and sure enough, yes I am haha). and it tracks (and graphs) your charging, discharging, idle, and charging on hold over time. It also tracks (and graphs) battery capacity and cycles. 
I would get this app and pay attention to the graph of your charging so you can see when you're charging, when you're gaining those cycles.

I mean, who knows. if you play around in the AlDente settings, sometimes it will forget sailing mode (if you use it) and go ahead and charge me up. Or maybe you have a faulty charger or cable that's cutting in and out. My phone charger is like that. sometimes if I pick up the phone, it stops charging.

**Comment by PearApple151** [comment]
> @davidwernhart Do you have an insight on this issue? My charging cycle number is increasing day by day, even when I plug in the cable most of the time with the help of AlDente to keep battery always at 80%.
> 
> Or, battery health can indeed still be greatly maintained even with large charging cycle number?

I would say yes. what's happening is not what we want. Something is amiss here. see my above response. but if you're charging from 50-80 3.333x a day and using the same number of cycles, you're still doing better than when you were charging 20-100 1.25x per day.

**Comment by PearApple151** [comment]
Did you turn off the optimized charging in your Mac's settings? I don't know if it would cause your problem, but I remember the AlDente guide mentioning that it could cause issues. doesn't hurt to check
